### ISSUE TITLE
cannot load library("cytofkit")

### ISSUE THREAD

**Issue by uchicagoxiao** [JinmiaoChenLab/cytofkit#23]
I used cytofkit before, and it worked well. But today, when I ran "library("cytofkit"), it showed as below:

Error: package or namespace load failed for ‘cytofkit’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/3.4/Resources/library/zoo/libs/zoo.so':
  `maximal number of DLLs reached...

How can I fix it? Thank you!

**Comment by MattMyint** [comment]
Hi,

You may want to [look at this issue regarding the max DLLs allowed](https://stackoverflow.com/a/43689526). as well as [the R documentation regarding OS-dependent DLL limits](https://www.rdocumentation.org/packages/base/versions/3.4.1/topics/dyn.load)

I did a quick check and found that loading cytofkit (and all its dependencies) loads 79 DLLs. Typically, cytofkit can be loaded alongside other packages.

If you reach the max number, its possible you have too many packages loaded at once in your current R session, in which case restarting the session then trying to load cytofkit again should work. Its also possible that you may have some open R sessions lying around, so it would be good to find if there are any unnecessary ones and close them.

If you really feel you need to keep other packages loaded concurrently, you can look at the links above to see how to increase the number of DLLs your machine can have loaded at a given time

**Comment by uchicagoxiao** [comment]
Thank you so much, Matt.

I followed your suggestion: closed all the packages in my R sessions, reset
DLL value, and restarted R. But when I loaded cytofkit, the result was the
same...

Xiao

**Comment by MattMyint** [comment]
Could you share the output of `sessionInfo()` and perhaps also some info like if you're on a 64bit machine and your machine's RAM?

**Comment by SamGG** [comment]
Just to be sure, if you run ```library(zoo)```, the command succeeds?

**Comment by uchicagoxiao** [comment]
Hi guys,

It works again! Last night, I deleted some packages that I recently installed, such as Seurat (I used seurat to analyze the single cell sequencing data). Before them, cytofkit worked very well in my computer. So I deleted them to see whether it can restore cytofit. After deletion, I restarted R, it still didn't work. But this morning (after overnight), when I tried again, cytofkit works again in my computer now. It's very weird though. 

Thank you all! Your suggestions are very helpful.
